catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/RefreshRegistrySubscriptionsTest.java

```diff
@@ -0,0 +1,390 @@
+
+    @Mock
+    private RegistryStore registryStore;
+
+    @Before
+    public void setUp() throws Exception {
+        when(parser.configureParser(anyList(), any(ClassLoader.class))).thenReturn(configurator);
+        when(filterBuilder.attribute(any(String.class))).thenReturn(attributeBuilder);
+        when(attributeBuilder.is()).thenReturn(expressionBuilder);
+        when(expressionBuilder.bool(any(Boolean.class))).thenReturn(getTestFilter());
+        refreshRegistrySubscriptions = spy(new RefreshRegistrySubscriptions());
+        federationAdminService.setRegistryTransformer(registryTransformer);
+        federationAdminService.setCatalogFramework(catalogFramework);
+        federationAdminService.setParser(parser);
+        federationAdminService.setFilterBuilder(filterBuilder);
+        refreshRegistrySubscriptions.setfederationAdminService(federationAdminService);
+        System.setProperty(SystemInfo.SITE_NAME, TEST_SITE_NAME);
+        System.setProperty(SystemInfo.VERSION, TEST_VERSION);
+        testMetacard = getPopulatedTestRegistryMetacard();
+
+    }
+
+    @Test
+    public void testRefreshRegistrySubscriptionsWhenPollableSourceIdsIAreEmpty() throws Exception {
+        refreshRegistrySubscriptions.refreshRegistrySubscriptions();
+        verify(refreshRegistrySubscriptions).refreshRegistrySubscriptions();
+    }
+
+    @Test
+    public void testcreateRemoteEntries() throws Exception {
+        Metacard firstMetacard = testMetacard;
+        Metacard secondMetacard = testMetacard;
+        Metacard thirdMetacard = getPopulatedTestRegistryMetacard();
+        thirdMetacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_ID,
+                RegistryObjectMetacardType.REGISTRY_ID + "1"));
+        QueryRequest request = getTestQueryRequest();
+        QueryResponse response = getPopulatedTestQueryResponse(request,
+                firstMetacard,
+                secondMetacard);
+        List<Metacard> localMetacards = new ArrayList<>();
+        localMetacards.add(firstMetacard);
+        localMetacards.add(secondMetacard);
+        doReturn(localMetacards).when(federationAdminService)
+                .getRegistryMetacards();
+        QueryRequest request1 = getTestQueryRequest();
+        QueryResponse response1 = getPopulatedTestQueryResponse(request1, thirdMetacard);
+        when(refreshRegistrySubscriptions.getBundleContext()).thenReturn(bundleContext);
+        when(bundleContext.getService(serviceReference)).thenReturn(registryStore);
+        when(registryStore.isPullAllowed()).thenReturn(true);
+        when(registryStore.getId()).thenReturn(TEST_ID);
+        when(catalogFramework.query(any(QueryRequest.class))).thenReturn(response, response1);
+        refreshRegistrySubscriptions.bindRegistryStore(serviceReference);
+        refreshRegistrySubscriptions.refreshRegistrySubscriptions();
+        //        verify(catalogFramework).create(any(CreateRequest.class));
```

Comment:
I'll have to think about these more. It is a little difficult to follow what is actually being tested since there are no assertions. Not sure if that is where you were headed with the commented out code in this test.